### PR TITLE
Issue #62 and #73: Update UG and DG for sort command and find command

### LINKED ISSUE #62: Update `find` command in DG and UG
Minor edits to `find` command in DG and UG on the compulsory/optional field parameters. 
Update the list of valid inputs for the `ATTRIBUTE_FIELD` parameter.

### LINKED ISSUE #73: Update UG and DG for `sort` command
Add `sort` command related details to user stories, use cases, features and command summary section of UG and DG.

### PR DESCRIPTION
**Pull request by charmainehly**
Currently, user stories, use cases and features do not contain
information and details pertaining to the sort command.

The information is missing for users and developers, who are unable
to attain consolidated information on the sort command
feature/functionality.

Let's update the UG and DG to include the sort command in the
following sections:
* Use cases (DeveloperGuide)
* User stories (DeveloperGuide)
* Features (UserGuide)
* Command summary (UserGuide)

Additional note: this commit also standardises the term
`applicant(s)` to `candidate(s)` in the UserGuide and DeveloperGuide.

<hr>

Quick Summary:
* Minor changes to: standardise `applicant(s)` to `candidate(s)`
* Minor edit to `find` command (closes #62 )
* Addition of `sort` command to DG/UG (closes #73 )

### CODE REVIEW

**Review by domlimm — COMMENTED** [review]
Small PR but lots of work done. Good job @charmainehly! LGTM 😄

@tiewweijian to approve and merge, thanks!

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -357,14 +358,15 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+1. User requests to find TAs containing specific keyword(s)
+2. TAlent Assistant™ returns a list of TAs that contain one or more of the given keyword(s) 
+in the specified attribute field
 
     Use case ends.
 
 **Extensions**
 
-* 2a. No TA applicant entries contain the given keyword
+* 2a. No TA candidate entries contains the given keyword in the specified attribute field
```
Comment (tiewweijian) [inline docs/DeveloperGuide.md:369]: Sorry something super minor here, but should the word contain be singular instead? (I might be wrong my proficiency in English is 👎🏻 )

**Review by tiewweijian — APPROVED** [review]
Just one minor thing that I flagged out. But I think we can fix it in the next iteration as well ! Thanks Charmaine for the good work, LGTM!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (charmainehly) [reply docs/DeveloperGuide.md:369]: No no, thanks for spotting I think you're right!! :) I'll make the quick change!

Comment (domlimm) [reply docs/DeveloperGuide.md]: I am confused 😖

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -273,13 +273,14 @@ interview availability dates.
...
-| `* * *`  | new user | view all available commands           | get familiarised with the system.                                             |
+| Priority | As a …​  | I want to …​                       | So that I can…​                                                               |
+|----------|----------|------------------------------------|-------------------------------------------------------------------------------|
+| `* * *`  | user     | add a candidate to the system      | keep track of their application details for consideration.                    |
+| `* * *`  | user     | delete a candidate from the system | remove entries that I no longer need.                                         |
+| `* * *`  | user     | find a candidate in the system     | access details of the candidate without having to go through the entire list. |
+| `* * *`  | user     | list all candidates in the system  | monitor the application pool.                                                 |
+| `*`      | user     | sort candidates                    | get familiarised with the system.                                             |
```
Comment (lzan98) [inline docs/DeveloperGuide.md:282]: Hi @charmainehly ! I was resolving my conflicts for #88 when I saw this. Do you want me to change `get familiarised with the system` for you? If so, what would you like me to change it to?

**Pull request merged**